components/core/src/clp_s/ArchiveReader.cpp

```diff
@@ -214,19 +206,36 @@ SchemaReader& ArchiveReader::create_schema_reader(
             should_marshal_records
     );
     auto timestamp_column_ids = m_timestamp_dict->get_authoritative_timestamp_column_ids();
-
     for (size_t i = 0; i < schema.size(); ++i) {
         int32_t column_id = schema[i];
         if (Schema::schema_entry_is_unordered_object(column_id)) {
             size_t length = Schema::get_unordered_object_length(column_id);
+
+            auto sub_schema = schema.get_view(i + 1, length);
+            auto mst_subtree_root_node_id = m_schema_tree->find_matching_subtree_root_in_subtree(
+                    -1,
+                    SchemaReader::get_first_column_in_span(sub_schema),
+                    Schema::get_unordered_object_type(column_id)
+            );
             append_unordered_reader_columns(
                     m_schema_reader,
-                    Schema::get_unordered_object_type(column_id),
-                    schema.get_view(i + 1, length),
+                    mst_subtree_root_node_id,
+                    sub_schema,
                     should_marshal_records
             );
             i += length;
             continue;
+        } else if (i >= schema.get_num_ordered()) {
```

Comment:
```suggestion
        } 
        if (i >= schema.get_num_ordered()) {
```
Do we need this branch? It seems that for this branch, we just add an entry to `m_global_id_to_unordered_object`, but we do nothing during serialization for this entry. So it is equivalent to not having this entry?

Reply:
Per sidebar we need it to populate local schema tree + to call mark_unordered_object.